frontend/src/components/SignUpForm/SignUpForm.jsx

```diff
@@ -2,74 +2,105 @@ import { useState, useContext } from "react";
 import "./SignUpForm.css"
 import { UserContext } from '../../UserContext.js';
 import { useNavigate, Link} from 'react-router-dom';
+import {validate as validateEmail} from 'email-validator';
 
 function SignUpForm (){
     const [email, setEmail] = useState("");
     const [password, setPassword] = useState("");
     const [firstName, setFirstName] = useState("");
     const [lastName, setLastName] = useState("");
 
+    const [emailError, setEmailError] = useState(null);
+    const [passwordError, setPasswordError] = useState(null);
+
     const { updateUser } = useContext(UserContext);
     const navigate = useNavigate();
 
     const handleOnSubmit = async (event) => {
         event.preventDefault();
-        event.stopPropagation();
-        const userObj = {
-            firstName: firstName,
-            lastName: lastName,
-            email: email,
-            password: password
-        }
+        if (firstName && lastName && email && password) {
+          const isEmailValid = validateEmail(email);
+          const isPasswordValid = (password.length >= 8);
+          // Validate email and password
+          if (!isEmailValid && !isPasswordValid){
+            setEmailError("Invalid email address");
+            setPasswordError("Password must be at least 8 characters long");
+            return;
+          }
```

Comment:
I don't think you need to handle this case separately, it would be handled by your individual checks for email and password below.